challenge_two/src/agent.spec.ts

```diff
@@ -0,0 +1,141 @@
+//   TOKEN2_ADDR: "0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599", // WBTC
+//   TOKEN2_VAL:  ethers.BigNumber.from("300"),
+//   TOKEN3_ADDR: "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2", // WETH
+//   TOKEN3_VAL: ethers.BigNumber.from("100"),
+//   POOL_ADDR2: "0x4585FE77225b41b697C938B018E2Ac67Ac5a20c0", // WBTC/WETH Pool
+//   Fee: ethers.BigNumber.from("1000"),
+// };
+
+const SWAP_TEST_2 = {
+  TOKEN2_ADDR: "0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599",
+  TOKEN2_VAL: ethers.BigNumber.from("100"),
+  TOKEN3_ADDR: "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
+  TOKEN3_VAL: ethers.BigNumber.from("400"),
+  POOL_ADDR2: "0x4585FE77225b41b697C938B018E2Ac67Ac5a20c0",
+  Fee: ethers.BigNumber.from("500"),
+};
+
+// These are also test vals for the swap event but can remain the same so we use these for both cases
+const SQRT_PRICE = ethers.BigNumber.from("10");
+const LIQ = ethers.BigNumber.from("1000");
+const TICK = ethers.BigNumber.from("1");
+
+describe("Uniswap Swaps Agent", () => {
+  let handleTransaction: HandleTransaction;
+
+  beforeAll(() => {
+    handleTransaction = agent.handleTransaction;
+  });
+
+  it("Should return empty findings if no events occur", async () => {
+    const mockTxEvent = new TestTransactionEvent();
+    const findings = await handleTransaction(mockTxEvent);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("Should return empty findings for non swap events ", async () => {
+    const mockTxEvent = new TestTransactionEvent();
+    mockTxEvent.addEventLog(TRANSFER_EVENT[0], createAddress("0x1"), [
+      createAddress("0x2"),
+      createAddress("0x1"),
+      5,
+    ]);
+    const findings = await handleTransaction(mockTxEvent);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("Should return empty findings for swaps not in Uniswap pool", async () => {
+    const mockTxEvent = new TestTransactionEvent();
+    mockTxEvent.addEventLog(SWAP_EVENT[0], createAddress("0x5"), [
+      SWAP_TEST_1.TOKEN0_ADDR,
+      SWAP_TEST_1.TOKEN1_ADDR,
+      SWAP_TEST_1.TOKEN0_VAL,
+      SWAP_TEST_1.TOKEN1_VAL,
+      SQRT_PRICE,
+      LIQ,
+      TICK,
+    ]);
+
+    mockTxEvent.setFrom(createAddress(createAddress("0x5")));
+    const findings = await handleTransaction(mockTxEvent);
+
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("Should return a finding for a swap in Uniswap Pool", async () => {
+    const mockTxEvent = new TestTransactionEvent();
+    mockTxEvent.addEventLog(SWAP_EVENT[0], SWAP_TEST_1.POOL_ADDR, [
+      SWAP_TEST_1.TOKEN0_ADDR,
+      SWAP_TEST_1.TOKEN1_ADDR,
+      SWAP_TEST_1.TOKEN0_VAL,
+      SWAP_TEST_1.TOKEN1_VAL,
+      SQRT_PRICE,
+      LIQ,
+      TICK,
+    ]);
+
+    const findings = await handleTransaction(mockTxEvent);
+    const swapFinding = createFinding(
+      SWAP_TEST_1.POOL_ADDR.toLocaleLowerCase()
+    );
+
+    expect(findings.length).toStrictEqual(1);
+    expect(findings[0]).toStrictEqual(swapFinding);
+  });
+
+  // REVIEWER PLEASE HELP?
```

Comment:
this will be solved after you implement the changes inside `agent.ts` I mentioned before.